style: Fix the stream and user list buttons on mobile in night mode.

skunkmb opened the pull request:
Currently, these buttons are displayed with a lighter background than other buttons. This PR updates their borders and background colors (along with the border on the search box) so that they match the night theme.

Fixes #10301.


**Screenshots:**

*Before:*

<img src="https://user-images.githubusercontent.com/17259768/44179868-3ba19900-a0ae-11e8-88c1-cf58f367a61c.png" alt="before" width="300px" />

*After:*

<img src="https://user-images.githubusercontent.com/17259768/44179876-3d6b5c80-a0ae-11e8-89d9-78e711b53352.png" alt="before" width="300px" />

synicalsyntax commented on static/styles/night_mode.scss:
```diff
@@ -236,6 +236,16 @@ on a dark background, and don't change the dark labels dark either. */
         color: hsl(0, 0%, 100%);
     }
 
+    #streamlist-toggle,
```
The colors of the icons should be changed too. Ideally, something like below:
![screenshot at aug 14 16-00-43](https://user-images.githubusercontent.com/15116870/44180290-61c83880-a0b0-11e8-8d27-65dc9459b5ae.png)

skunkmb replied:
@synicalsyntax OK, I've updated the color of the two icons to be the same as the home icon (`#dddeee`). I also edited the initial PR comment with a new screenshot showing this change. Please let me know if there is anything else that I should change or fix. Thanks!

synicalsyntax replied:
I don't see a double border with the following git diff, but correct me if I'm wrong:

```diff
diff --git a/static/styles/night_mode.scss b/static/styles/night_mode.scss
index f8263670d..2f7b09c49 100644
--- a/static/styles/night_mode.scss
+++ b/static/styles/night_mode.scss
@@ -76,6 +76,8 @@ on a dark background, and don't change the dark labels dark either. */
 
     .new-style .button:not(.save-status),
     .new-style .button[disabled=disabled]:hover,
+    #streamlist-toggle-button,
+    #userlist-toggle-button,
     input[type="text"],
     input[type="email"],
     input[type="password"],
diff --git a/static/styles/right-sidebar.scss b/static/styles/right-sidebar.scssindex 6c384f8b9..5b9c2fd8b 100644
--- a/static/styles/right-sidebar.scss
+++ b/static/styles/right-sidebar.scss
@@ -186,12 +186,12 @@
     right: 0px;
     text-align: center;
     vertical-align: middle;
-    border-left: 2px solid hsl(204, 20%, 74%);
 }
 
 #userlist-toggle-button {
     text-decoration: none;
     background-color: hsl(0, 0%, 89%);
+    border-left: 2px solid hsl(204, 20%, 74%);
     color: hsl(0, 0%, 52%);
     display: block;
     width: 40px;
diff --git a/static/styles/zulip.scss b/static/styles/zulip.scss
index 4178fe3f3..ba5464333 100644
--- a/static/styles/zulip.scss
+++ b/static/styles/zulip.scss
@@ -1689,7 +1689,6 @@ blockquote p {
     left: 0px;
     text-align: center;
     vertical-align: middle;
-    border-right: 2px solid hsl(204, 20%, 74%);
 }
 
 #streamlist-toggle-button {
@@ -1698,6 +1697,7 @@ blockquote p {
     display: block;
     position: relative;
     background-color: hsl(0, 0%, 89%);
+    border-right: 2px solid hsl(204, 20%, 74%);
     width: 40px;
     height: 19px;
     padding-top: 12px;
(END)
```
![screenshot at aug 14 16-00-43](https://user-images.githubusercontent.com/15116870/44232449-ee790200-a155-11e8-9916-82c104d8ad3f.png)

synicalsyntax commented on static/styles/night_mode.scss:
```diff
@@ -236,6 +236,17 @@ on a dark background, and don't change the dark labels dark either. */
+
+    #streamlist-toggle-button,
+    #userlist-toggle-button {
+        color: #dddeee;
```
Can you use HSL instead of hex? Also, you might as well use `color: inherit` and move the border-color declarations to the toggle buttons themselves instead of the parent elements so you don't have to change the border color in a separate selector.

skunkmb commented:
@synicalsyntax OK, I changed the `toggle-button`s to use `color: inherit`. I'm not able to give them the border color declarations, though, since those need to override the already-existing declarations for the `toggle`s. (Adding it to the `toggle-button`s and removing it from the `toggle`s creates a double-border.)

Please let me know your thoughts. Thanks!

synicalsyntax commented on static/styles/night_mode.scss:
```diff
@@ -266,6 +277,7 @@ on a dark background, and don't change the dark labels dark either. */
     }
 
     #searchbox,
+    #searchbox_legacy,
```
Is there a reason why this was added here?

skunkmb commented:
@synicalsyntax Ah, I didn't know that you meant to edit the CSS of the default styles. I've updated the PR to that it moves the border declarations to those `toggle-button`s. I still had to define the night mode button styles separately, though (instead of adding them to the button selectors you showed in your diff at the beginning). If I didn't, there was an area to the right of the button that was incorrectly lightened:

<img width="90" alt="screen shot 2018-08-16 at 4 25 47 pm" src="https://user-images.githubusercontent.com/17259768/44240390-aae0c100-a172-11e8-879c-9eeefc6b7dad.png">

As for your other comment, I added the `#searchbox_legacy` line so that when the legacy searchbox is used on mobile in night mode, it also gets the updated background color and border color that the new one gets.

Please let me know your thoughts. Thanks for giving me feedback on this!

synicalsyntax commented:
@skunkmb Regarding the legacy searchbox changes: If you're only trying to change the legacy searchbox's background/border color, can you add those properties to the `#searchbox_legacy .navbar-search` selector we already have?

![screenshot at aug 17 11-42-09](https://user-images.githubusercontent.com/15116870/44283264-a79d1200-a212-11e8-98c6-6304e7df3528.png)


Also, I can't reproduce the incorrectly-colored background color issue you're facing (I used the selectors in my original suggestion in the previous screenshot). Can you rebase your branch off of latest master and see if that changes anything?

skunkmb commented:
@synicalsyntax OK, I moved it to the `.navbar-search` selector. The incorrect background with your diff still persists, though, for me (even after a rebase). For me, the issue is visible at around `950px` of width; try making your browser that wide and check the top right. (To clarify, what I'm seeing is that at `950px` of width, the button at the top right has a "split-in-two" look due to the transparent background.)

<img width="949" alt="screen shot 2018-08-17 at 3 22 14 pm" src="https://user-images.githubusercontent.com/17259768/44291275-32d9d000-a232-11e8-9bc5-bf07e87be2ed.png">


I apologize if you can't reproduce what I'm seeing, but for now I'm going to stick with the separate selector that doesn't produce the issue for me. In this PR, I currently have a version that is working for me.

Please let me know if you are able to reproduce what I am seeing in your diff and your thoughts on this change. Thank you!

skunkmb commented:
I'm closing this since it's gotten a bit stale, and I think that there's potentially a more straightforward way to solve this.
